core/src/main/java/io/parsingdata/metal/token/Def.java

```diff
@@ -66,7 +66,7 @@ public Def(final String name, final ValueExpression size, final Encoding encodin
         }
         return environment
             .slice(dataSize)
-            .map(slice -> success(environment.add(new ParseValue(scope, this, slice, encoding)).seek(dataSize.add(environment.offset))))
+            .map(slice -> environment.add(new ParseValue(scope, this, slice, encoding)).seek(dataSize.add(environment.offset)))
```

Comment:
Wait, what? How come this `success` can be safely deleted here?

Reply:
Because ```Environment#add``` now returns an optional environment.

Reply:
Because ```Environment#seek``` now returns an optional environment.

Reply:
That would lead to a very redundant `.map(x -> x).orElse(empty())` construct... So I appreciate the sentiment but I don't really know what a good solution is...

Reply:
WIsn't something like the following possible:
```java
return environment
    .slice(dataSize)
    .map(slice -> environment.add(new ParseValue(scope, this, slice, encoding)))
    .map(env -> env.seek(dataSize.add(environment.offset)))
    .orElse(failure());
```
Not saying it is the better solution, but I think it is functionally possible? Perhaps a bit more clear?

Reply:
Sure, but then something like `Environment.add()` would have to return an `Optional` even though there's no real reason it could fail. I see the relative beauty of writing it like this, but implementing brings all kinds of trade-offs.

Reply:
Or I am overlooking something, because to me it seems like you are confusing ```map``` with ```flatMap``` :smile: